processor/src/main/java/org/mapstruct/ap/internal/util/MapperConfiguration.java

```diff
@@ -155,6 +156,15 @@ public NullValueCheckStrategyPrism getNullValueCheckStrategy() {
         }
     }
 
+    public InjectionStrategyPrism getInjectionStrategy() {
```

Comment:
@filiphr No problem, I appreciate any feedback :) I will implement the review feedback soon

Regarding the tests: I agree, the easiest way is to copy the existing mappers and once the field injection is removed in a later release, we can just delete those mappers.

Regarding the injection of additional fields: The easiest way would be to not handle this case at all and possibly mix constructor with setter injection. However, this would also impact testability. A user would have to call a constructor and call the additional setters. I can't come up with an easy way of handling the injection of additional fields, though.

Regarding the decorators: If I understand you correctly, this is also implemented, see the example implementations in my first post. Otherwise, could you please give a sample mapper and the supposed implementation that MapStruct will generate?

Maybe another opinion regarding tests and the injection of additional fields from @gunnarmorling  @agudian @sjaakd would be nice :)